pages/repo/edit.js

```diff
@@ -861,6 +905,14 @@ export default compose(
       }
     }
   }),
+  graphql(getTemplateById, {
+    skip: ({ router }) => !router.query.templateRepoId,
```

Comment:
We need to set [`config.name`](https://www.apollographql.com/docs/react/api/react/hoc/#configname) here to avoid a collision with the other queries:
```suggestion
    name: 'templateData',
    skip: ({ router }) => !router.query.templateRepoId,
```
(the others don't do it because always only one runs, but here it could run in combo with one of the others)

And then we need to read the data from `props.templateData`.

Make sure to also duplicate the early return for loading and errors of `props.templateData` in `loadState` (line ~420) and add `templateData.error` to the possible display error variable in `render` (line ~709)

Reply:
Actually, do we have name collision? I just remember that I renamed `repo`to `templateRepo`in the query (l.104). Shouldn't that be enough or am I missing something?

Reply:
Renaming in the query isn't enough because the data objects are not automatically merged by apollo. Merging would also lead to ambiguity regarding `loading` and `error`